test/integration/job/job_test.go

```diff
@@ -1667,6 +1667,165 @@ func TestIndexedJob(t *testing.T) {
 	validateTerminatedPodsTrackingFinalizerMetric(t, 5)
 }
 
+func TestJobPodReplacementPolicy(t *testing.T) {
+	const initialCompletions int32 = 2
+	var podReplacementPolicy = func(obj batchv1.PodReplacementPolicy) *batchv1.PodReplacementPolicy {
+		return &obj
+	}
+	var completionPolicy = func(obj batchv1.CompletionMode) *batchv1.CompletionMode {
+		return &obj
+	}
+	jobSpecIndexedDefault := &batchv1.JobSpec{
+		Parallelism:    pointer.Int32Ptr(initialCompletions),
+		Completions:    pointer.Int32Ptr(initialCompletions),
+		CompletionMode: completionPolicy(batchv1.IndexedCompletion),
+	}
+	cases := map[string]struct {
+		podReplacementPolicyEnabled bool
+		wantTerminating             *int32
+		wantFailed                  int
+		wantActive                  int
+		jobSpec                     *batchv1.JobSpec
+	}{
+		"feature flag off, delete pods and verify no terminating status": {
+			jobSpec:    jobSpecIndexedDefault,
+			wantActive: 2,
+			wantFailed: 2,
+		},
+		"feature flag true, delete pods and verify terminating status": {
+			podReplacementPolicyEnabled: true,
+			wantTerminating:             pointer.Int32(2),
```

Comment:
nit: I would suggest moving the `want` to the bottom consistently for better readability.

Reply:
Applied your suggestion.